runtimevar/constantvar/constantvar.go

```diff
@@ -37,21 +36,30 @@ func NewError(err error) *runtimevar.Variable {
 	return runtimevar.New(&watcher{err: err})
 }
 
-// watcher implements driver.Watcher.
+// watcher implements driver.Watcher and driver.State.
 type watcher struct {
 	value interface{}
 	err   error
 	t     time.Time
 }
 
-func (w *watcher) WatchVariable(ctx context.Context, prevVersion interface{}, prevErr error) (*driver.Variable, interface{}, time.Duration, error) {
+func (w *watcher) Value() (interface{}, error) {
+	return w.value, w.err
+}
+
+func (w *watcher) UpdateTime() time.Time {
+	return w.t
+}
+
+func (w *watcher) WatchVariable(ctx context.Context, prev driver.State) (driver.State, time.Duration) {
 
 	// The first time this is called, return the constant value.
-	if prevVersion == nil && prevErr == nil {
-		return &driver.Variable{Value: w.value, UpdateTime: w.t}, true, 0, w.err
+	if prev == nil {
+		return w, 0
 	}
 	// On subsequent calls, block ~forever as the value will never change.
-	return nil, nil, time.Duration(math.Inf(+1)), nil
+	<-ctx.Done()
+	return nil, 0
```

Comment:
GitHub won't let me carry on the previous thread:

I see. This is subtle enough that I would call it out in a comment: the concrete type will be interrupted in its wait by `Context.Done`. I am a bit worried that a refactoring of the concrete type might accidentally "optimize" this out, but up to you.

Reply:
I guess it's better to be explicit. I changed it to return itself with err set to ctx.Err().